schema/v2/card_schema.json

```diff
@@ -0,0 +1,328 @@
+{
+  "$id": "https://github.com/netrunnerdb/netrunner-cards-json/blob/master/schema/v2/card_schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "additionalProperties": false,
+  "allOf": [
+    {
+      "title": "Faction id must be valid for the side.",
+      "type": "object",
+      "oneOf": [
+        {
+          "properties": {
+            "side_id": {
+              "const": "corp"
+            },
+            "faction_id": {
+              "enum": [
+                "haas-bioroid",
+                "jinteki",
+                "nbn",
+                "neutral-corp",
+                "weyland-consortium"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "side_id": {
+              "const": "runner"
+            },
+            "faction_id": {
+              "enum": [
+                "adam",
+                "anarch",
+                "apex",
+                "criminal",
+                "neutral-runner",
+                "shaper",
+                "sunny-lebeau"
+              ]
+            }
+          }
+        }
+      ]
+    },
+    {
+      "title": "Type codes must be valid for the side.",
+      "type": "object",
+      "oneOf": [
+        {
+          "properties": {
+            "side_id": {
+              "const": "corp"
+            },
+            "card_type_id": {
+              "enum": [
+                "agenda",
+                "asset",
+                "ice",
+                "corp_identity",
+                "operation",
+                "upgrade"
+              ]
+            }
+          }
+        },
+        {
+          "properties": {
+            "side_id": {
+              "const": "runner"
+            },
+            "card_type_id": {
+              "enum": [
+                "event",
+                "hardware",
+                "runner_identity",
```

Comment:
I'm glad we're separating these types, but this isn't consistent with the faction equivalent (`neutral-runner`, `neutral-corp`).

(Also note the hyphen/underscore discrepancy)

Reply:
what is inconsistent?  Is it just the underscores?  I was leaving off broader underscore-ification to it's own PR, but included this one as in scope for the verification in this schema.

Reply:
Corp/runner goes first in the identity enum but second in the faction enum. It's a super minor thing but something that's inevitably going to trip me up some day.

Reply:
i didn't actually follow that, but i did sort the enums.

oh, are you talking about neutral-{corp,runner} vs. {corp,runner}_identity ?  I actually think that is natural.  We don't say "an identity that is a corp" we say "a corp id" and likewise, we don't say "a corp card that is neutral" but "a neutral corp card" typically.

Hopefully solid tests help prevent mixups thoroughly!